src/main/java/com/foo/gosucatcher/domain/image/infrastructure/FileSystemExpertImageService.java

```diff
@@ -0,0 +1,139 @@
+
+@RequiredArgsConstructor
+@Service
+public class FileSystemExpertImageService implements ImageService {
+
+	private final ExpertImageRepository expertImageRepository;
+	private final ExpertRepository expertRepository;
+
+	// @Value("${spring.servlet.multipart.location}")
+	private String uploadPath;
+
+	@Override
+	public String store(ImageUploadRequest request) {
+		try {
+			MultipartFile file = request.file();
+			if (file.isEmpty()) {
+				throw new InvalidValueException(ErrorCode.INVALID_IMAGE);
+			}
+
+			Path root = Paths.get(uploadPath, request.id().toString());
+			if (!Files.exists(root)) {
+				Files.createDirectories(root);
+			}
+
+			String originalFilename = file.getOriginalFilename();
+			String extension = FilenameUtils.getExtension(originalFilename);
+			String newFilename = UUID.randomUUID() + "." + extension;
+
+			try (InputStream inputStream = file.getInputStream()) {
+				Files.copy(inputStream, root.resolve(newFilename), StandardCopyOption.REPLACE_EXISTING);
+			}
+
+			Expert expert = expertRepository.findById(request.id())
+				.orElseThrow(() -> new EntityNotFoundException(ErrorCode.NOT_FOUND_EXPERT));
+
+			ExpertImage expertImage = ExpertImage.builder()
+				.filename(newFilename)
+				.url(root.resolve(newFilename).toString())
+				.size(file.getSize())
+				.expert(expert)
+				.build();
+
+			expertImageRepository.save(expertImage);
+			return newFilename;
+		} catch (EntityNotFoundException e) {
+			throw e;
+		} catch (Exception e) {
+			throw new BusinessException(ErrorCode.INTERNAL_SERVER_ERROR);
+		}
+	}
+
+	@Override
+	public Stream<Path> loadAll(Long expertId) {
+		try {
+			Path root = Paths.get(uploadPath, expertId.toString());
+			return Files.walk(root, 1)
+				.filter(path -> !path.equals(root));
+		} catch (IOException e) {
+			throw new ExpertImageIOException(ErrorCode.INTERNAL_SERVER_ERROR);
+		}
+	}
+
+	@Override
+	public Path load(Long expertId, String filename) {
+		return Paths.get(uploadPath, expertId.toString()).resolve(filename);
+	}
+
+	@Override
+	public Resource loadAsResource(Long expertId, String filename) {
+		try {
+			Path file = load(expertId, filename);
+			Resource resource = new UrlResource(file.toUri());
+			if (resource.exists() || resource.isReadable()) {
+				return resource;
+			} else {
+				throw new ExpertImageIOException(ErrorCode.INVALID_IMAGE);
+			}
+		} catch (MalformedURLException e) {
+			throw new ExpertImageIOException(ErrorCode.INVALID_IMAGE);
+		}
+	}
+
+	@Override
+	public void delete(Long expertId, String filename) {
+		try {
+			Path root = Paths.get(uploadPath, expertId.toString());
+
+			if (!Files.exists(root)) {
+				throw new EntityNotFoundException(ErrorCode.NOT_FOUND_IMAGE);
+			}
+
+			Optional<ExpertImage> expertImage = expertImageRepository.findByFilename(filename);
+
+			if (expertImage.isPresent()) {
+				expertImageRepository.delete(expertImage.get());
+			} else {
+				throw new EntityNotFoundException(ErrorCode.NOT_FOUND_IMAGE);
+			}
```

Comment:
if문 내에서 `isPresent`로 확인 후에 이런식으로 작업하는게 안티 패턴이라고 들어서 
Optional의 메소드를 활용하는 것도 좋을 것 같아요!

```suggestion
			expertImage.ifPresentOrElse (
				expertImage -> expertImageRepository.delete(expertImage.get()),
			() -> {
				throw new EntityNotFoundException(ErrorCode.NOT_FOUND_IMAGE);
			}
```